forc/src/utils/cli_error.rs

```diff
@@ -1,91 +1,71 @@
-        let message = format!(
-            "could not find `{}` in `{}` or any parent directory",
-            MANIFEST_FILE_NAME,
-            curr_dir.display()
-        );
-        Self { message }
-    }
-
-    pub fn parsing_failed(project_name: &str, errors: Vec<CompileError>) -> Self {
-        let message = errors
-            .iter()
-            .map(|e| e.to_friendly_error_string())
-            .collect::<Vec<String>>()
-            .join("\n");
-
-        Self {
-            message: format!("Parsing {} failed: \n{}", project_name, message),
-        }
-    }
-
-    pub fn wrong_sway_type(project_name: &str, wanted_type: &str, parse_type: &str) -> Self {
-        let message = format!(
-            "{} is not a '{}' it is a '{}'",
-            project_name, wanted_type, parse_type
-        );
-        Self { message }
-    }
-
-    pub fn fuel_core_not_running(node_url: &str) -> Self {
-        let message = format!("could not get a response from node at the URL {}. Start a node with `fuel-core`. See https://github.com/FuelLabs/fuel-core#running for more information", node_url);
-        Self { message }
-    }
-}
-
-impl fmt::Display for CliError {
-    fn fmt(&self, f: &mut fmt::Formatter) -> Result<(), fmt::Error> {
-        write!(f, "{}", self)
-    }
+pub fn parsing_failed(project_name: &str, errors: Vec<CompileError>) -> anyhow::Error {
+    let error = errors
+        .iter()
+        .map(|e| e.to_friendly_error_string())
+        .collect::<Vec<String>>()
+        .join("\n");
+    let message = format!("Parsing {} failed: \n{}", project_name, error);
+    Error::msg(message)
 }
 
-impl From<&str> for CliError {
-    fn from(s: &str) -> Self {
-        CliError {
-            message: s.to_string(),
-        }
-    }
+pub fn wrong_project_type(
+    project_name: &str,
+    wanted_type: &str,
+    parse_type: &str,
+) -> anyhow::Error {
+    let message = format!(
+        "{} is not a '{}' it is a '{}'",
+        project_name, wanted_type, parse_type
+    );
+    Error::msg(message)
 }
 
-impl From<String> for CliError {
-    fn from(s: String) -> Self {
-        CliError { message: s }
-    }
+pub fn fuel_core_not_running(node_url: &str) -> anyhow::Error {
+    let message = format!("could not get a response from node at the URL {}. Start a node with `fuel-core`. See https://github.com/FuelLabs/fuel-core#running for more information", node_url);
+    Error::msg(message)
 }
 
-impl From<io::Error> for CliError {
-    fn from(e: io::Error) -> Self {
-        CliError {
-            message: e.to_string(),
-        }
-    }
-}
-
-impl From<AddrParseError> for CliError {
-    fn from(e: AddrParseError) -> Self {
-        CliError {
-            message: e.to_string(),
-        }
-    }
-}
+/// Given the current directory and file type, determines whether the correct project type is present.
+pub fn check_project_type(curr_dir: PathBuf, wanted_type: &str) -> Result<Manifest> {
```

Comment:
I actually found a way to make this work with `TreeType`, it does end up giving you _more_ info if you happen to give it a Library, but I actually think that's cleaner than recreating a second enum with essentially the same properties (feel free to dispute that), another problem with that is that we would be trying to match two different types. One of `TreeType` and one of `ProgramType`.

Reply:
To avoid recreating the `manifest` so many times I've gone ahead and created something of a compromise from your above code and my previous code. I think this feels much nicer than before, I also moved the related functions from `cli_error` to `forc-pkg::pkg` as you previously instructed.